### ISSUE TITLE
bad flow typing with empty block

### ISSUE THREAD

**Issue by lewis6991** [EmmyLuaLs/emmylua-analyzer-rust#620]
```lua
local a --- @type integer?

if not a then
end

--- @type integer should warn
local _ = a
```
